<?xml version="1.0" ?>

<launch>
    <arg name="task_file" default="$(find quad_controllers)/config/mpc/task.info"/>
    <arg name="reference_file" default="$(find quad_controllers)/config/command/reference.info"/>
    <arg name="urdf_file" default="$(find unitree_description)/urdf/a1.urdf"/>

    <arg name="gaitCommandFile" default="$(find quad_controllers)/config/command/gait.info"/>

    <!-- make the files into global parameters -->
    <param name="task_file" value="$(arg task_file)"/>
    <param name="referenceFile" value="$(arg reference_file)"/>
    <param name="urdf_file" value="$(arg urdf_file)"/>
    <param name="gaitCommandFile" value="$(arg gaitCommandFile)"/>
    <rosparam file="$(find quad_controllers)/config/default.yaml" command="load"/>

    <node name="controller_loader" pkg="controller_manager" type="controller_manager" output="screen"
          args="load
          controllers/joint_state_controller
          controllers/imu_sensor_controller
          controllers/ocs2_controller
"/>

    <node pkg="ocs2_legged_robot_ros" type="legged_robot_target" name="legged_robot_target"
          output="screen" launch-prefix="gnome-terminal --"/>

    <node pkg="ocs2_legged_robot_ros" type="legged_robot_gait_command" name="legged_robot_gait_command"
          output="screen" launch-prefix="gnome-terminal --"/>
</launch>
